readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/writer/PartitionedWriter.java

```diff
@@ -285,7 +285,7 @@ public Expirer(Collection<PartitionedWriter<MESSAGE_KIND>> writers, TemporalAmou
         public void start(Thread.UncaughtExceptionHandler uncaughtExceptionHandler, String name) {
             runningThread = new Thread(() -> {
                 while (!Thread.currentThread().isInterrupted()) {
-                    writers.forEach(PartitionedWriter::expireConsumers);
+                    run();
```

Comment:
Same remark than for the Heartbeat consumer + we should also remove all synchronization on the hashmap of PartitionedWriter

Reply:
done